codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlMemberDeserVisitor.java

```diff
@@ -0,0 +1,370 @@
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ResourceShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.SetShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeVisitor;
+import software.amazon.smithy.model.shapes.ShortShape;
+import software.amazon.smithy.model.shapes.StringShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.shapes.TimestampShape;
+import software.amazon.smithy.model.shapes.UnionShape;
+import software.amazon.smithy.model.traits.EnumTrait;
+import software.amazon.smithy.model.traits.TimestampFormatTrait.Format;
+
+/**
+ * Visitor to generate member values for aggregate types deserialized from documents.
+ */
+public class XmlMemberDeserVisitor implements ShapeVisitor<Void> {
+    private final GenerationContext context;
+    private final String dataDest;
+    private final Format timestampFormat;
+    private final Predicate<Shape> filter;
+    private final Boolean isFlattened;
+
+
+    public XmlMemberDeserVisitor(GenerationContext context, String dataDest, Format timestampFormat, Predicate<Shape> filter, Boolean isFlattened) {
+        this.context = context;
+        this.dataDest = dataDest;
+        this.timestampFormat = timestampFormat;
+        this.filter = filter;
+        this.isFlattened = isFlattened;
+    }
+
+    @Override
+    public Void blobShape(BlobShape shape) {
+        GoWriter writer = context.getWriter();
+        writer.write("var data string");
+        writer.write("err := decoder.Decoder.DecodeElement(&data, &t)");
+        writer.write("if err != nil { return err }");
+        writer.insertTrailingNewline();
+
+        writer.addUseImports(SmithyGoDependency.BASE64);
+        writer.write("$L, err = base64.StdEncoding.DecodeString(data)", dataDest);
+        writer.write("if err != nil { return err }");
+        return null;
+    }
+
+    @Override
+    public Void booleanShape(BooleanShape shape) {
+        GoWriter writer = context.getWriter();
+        writer.addUseImports(SmithyGoDependency.FMT);
+        consumeToken(shape);
+
+        writer.openBlock("if val != nil {", "}", () -> {
+            writer.addUseImports(SmithyGoDependency.STRCONV);
+            writer.write("xtv, err := strconv.ParseBool(string(val))");
+            writer.openBlock("if err != nil {", "}", () -> {
+                writer.write("return fmt.Errorf(\"expected $L to be of type *bool, got %T instead\", val)",
+                        shape.getId().getName());
+            });
+            writer.write("$L = &xtv", dataDest);
+        });
+        return null;
+    }
+
+    /**
+     * Consumes a single token into the variable "val", returning on any error.
+     */
+    private void consumeToken(Shape shape) {
+        if (!filter.test(shape)) {
+         return;
```

Comment:
Indentation. Also, won't this result in `val` not being defined at all and thus breaking the various nil checks that rely on it?

Reply:
Will try to clarify the usage or extract the val variable out (if possible).